### ISSUE TITLE
Margin for search button

### ISSUE THREAD

**Issue by axel1200** [web-cat/code-workout#170]
![Screenshot from 2020-08-07 18-30-26](https://user-images.githubusercontent.com/19996713/89667255-1aaa6200-d8dc-11ea-91c9-643a650f852f.png)
Would expect 9px margin like the other buttons in navbar

**Comment by s-edwards** [comment]
Actually, I think the search button plus search box should appear as one control, as is typical on most websites. Here, the rounded corners of the search button shouldn't appear rounded on the left edge, and the corners of the text field shouldn't be rounded on the right, so that they appear as one control. In the current structure, these are separate inputs, but we should switch to bootstrap's "add-on button" approach and make the button itself an add-on to the search field, as shown here: https://getbootstrap.com/docs/4.0/components/input-group/.

**Comment by axel1200** [comment]
Gonna do that. That does also exist in bootstrap 3 so after reading how haml works that should be an easy thing to do

**Comment by s-edwards** [comment]
Thanks, appreciate it!

**Comment by s-edwards** [comment]
Well, the revision loses the sizing from the previous version, causing the search field to expand to full-width on the nav bar and splitting the nav bar into 3 rows with the search box filling the middle row. Not immediately clear from the CSS what is causing this? We need to fix that before we can deploy the fix.

**Comment by ayaankazerouni** [comment]
I'm not sure what you mean. Can you add a screenshot? This is what I see (full width):
![image](https://user-images.githubusercontent.com/11241828/89849109-9f93b680-db3c-11ea-9620-ca69c2a1cc3f.png)

**Comment by ayaankazerouni** [comment]
Ah, ok, Chrome is borked 
![image](https://user-images.githubusercontent.com/11241828/89849381-42e4cb80-db3d-11ea-9ae5-f43250d8e978.png)

**Comment by ayaankazerouni** [comment]
Ok, it appears that **Firefox** is borked, because Safari and Chrome render this the same way, but Firefox renders it wrong.

**Comment by s-edwards** [comment]
Yeah, I about have it fixed. preparing to commit a change to staging to repair this, which is due to the use of table-style display properties on the input group. No worries--I think I can fix it pretty easy.

**Comment by s-edwards** [comment]
Basically, if it doesn't work on chrome or safari, then it doesn't work ;-).

**Comment by s-edwards** [comment]
OK, I just checked in a commit to staging that fixes this for chrome and safari. Can you test on firefox to see if it is still working there? Also, I tried to add a fix for the left sidebar scrolling background issue, but am leaving testing for Ayaan :-). If you can check these out, then we can close the issue and merge to master.

**Comment by ayaankazerouni** [comment]
Search bar looks good on Chrome, Firefox, and Safari.

For the left sidebar background—are you talking about the workout form? That issue still persists on all browsers 😕

**Comment by axel1200** [comment]
My fault used Firefox :(

**Comment by axel1200** [comment]
Probably also  reason to use BS 4 works better cross browser

**Comment by s-edwards** [comment]
Sorry to hear the left sidebar background didn't get fixed. Do we have a separate issue for that? Also, can you point me to a specific workout where this is exhibited in the edit form?

My belief is that the issue always exists in the sidebar, not just on the edit form. However, the sidebar has to have "enough" content in it for the issue to be visible, and search results in the edit form is the easiest way to generate that volume.

It sounds like this issue is addressed, and BS 4 (as Alex mentions) is already a separate issue, so we can close this one if we have a separate issue for the sidebar background scrolling. Glad this one is fixed, though.

**Comment by ayaankazerouni** [comment]
> Do we have a separate issue for that? Also, can you point me to a specific workout where this is exhibited in the edit form?

#174 

> My belief is that the issue always exists in the sidebar, not just on the edit form. However, the sidebar has to have "enough" content in it for the issue to be visible, and search results in the edit form is the easiest way to generate that volume.

Yeah. I've seen it in some giant workouts that have enough exercises in them for the issue to manifest.
